screeps-game-api/src/game.rs

```diff
@@ -236,29 +236,34 @@ pub mod map {
             unsafe { mem::transmute(callback_type_erased) };
 
         FR_CALLBACK.set(&callback_lifetime_erased, || {
-            let v = js!(return Game.map.findRoute(@{from_room.as_ref()}, @{to_room.as_ref()}, @{callback}););
+            let v = js!(return Game.map.findRoute(@{from_room}, @{to_room}, @{callback}););
             parse_find_route_returned_value(v)
         })
     }
 
-    fn parse_find_route_returned_value(v: Value) -> Result<Vec<ExitDirection>, ReturnCode> {
+    fn parse_find_route_returned_value(v: Value) -> Result<Vec<RoomRouteStep>, ReturnCode> {
         match v {
             Value::Number(x) => {
                 let i: i32 = x.try_into().unwrap();
-                Err(i.try_into().expect("Unexpected return code."))
+                Err(i
+                    .try_into()
+                    .expect(&format!("Unexpected return code: {:?}.", i)))
```

Comment:
I think if we want to include a message here, it would be better to `match` on the result or use `.unwrap_or_else(|_| panic!())` rather than use `.expect()`. Using `format!()` like this will allocate a new string and do string formatting even if the result is `Ok()`.